### PR TITLE
Detect cassandra leak

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by remk** [comment]
```
[a03a8ac05b4ba6766bce9183948e94691288317d] [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.02 s - in org.apache.james.backends.cassandra.init.CassandraConfigurationReadingTest
[a03a8ac05b4ba6766bce9183948e94691288317d] [INFO] 
[a03a8ac05b4ba6766bce9183948e94691288317d] [INFO] Results:
[a03a8ac05b4ba6766bce9183948e94691288317d] [INFO] 
[a03a8ac05b4ba6766bce9183948e94691288317d] [ERROR] Errors: 
[a03a8ac05b4ba6766bce9183948e94691288317d] [ERROR]   CassandraTableManagerTest » IllegalState Cluster already running
[a03a8ac05b4ba6766bce9183948e94691288317d] [ERROR]   CassandraTypeProviderTest » IllegalState Cluster already running
[a03a8ac05b4ba6766bce9183948e94691288317d] [ERROR]   CassandraTypesCreatorTest » IllegalState Cluster already running
[a03a8ac05b4ba6766bce9183948e94691288317d] [ERROR]   CassandraHealthCheckTest.setUp:39 » IllegalState Cluster already running
[a03a8ac05b4ba6766bce9183948e94691288317d] [ERROR]   CassandraHealthCheckTest.setUp:39 » IllegalState Cluster already running
[a03a8ac05b4ba6766bce9183948e94691288317d] [INFO] 
```

**Comment by mbaechler** [comment]
please copy/paste the stack traces for this PR because they contain the root cause

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraClusterTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class CassandraClusterTest {
+
+    @RegisterExtension
+    static DockerCassandraExtension cassandraExtension = new DockerCassandraExtension();
+    CassandraCluster cnx1;
```
Comment (rouazana) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraClusterTest.java]: connection?

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java [hunk 2]
```diff
@@ -33,13 +35,23 @@
...
     private final CassandraModule module;
     private Session session;
     private CassandraTypesProvider typesProvider;
     private Cluster cluster;
 
     public static CassandraCluster create(CassandraModule module, Host host) {
-        return new CassandraCluster(module, host);
+        assertClusterNotRunning();
+        CassandraCluster cassandraCluster = new CassandraCluster(module, host);
+        startStackTrace = Optional.of(new Exception());
```
Comment (rouazana) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java]: can we add a description to this exception to explain that it is the root cause, meaning where the first cluster was launched?

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/utils/CassandraHealthCheckTest.java [hunk 3]
```diff
@@ -33,13 +34,19 @@
...
-        CassandraCluster cassandra = CassandraCluster.create(CassandraModule.builder().build(), cassandraServer.getHost());
+        cassandra = CassandraCluster.create(CassandraModule.builder().build(), cassandraServer.getHost());
         healthCheck = new CassandraHealthCheck(cassandra.getConf());
     }
 
+    @AfterEach
+    void tearDown() {
+        cassandra.close();
+    }
+
```
Comment (rouazana) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/utils/CassandraHealthCheckTest.java]: this commit should be a fixup? (e51b1022963fbe4df10a4444dbe737549821db7c)

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraClusterTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package org.apache.james.backends.cassandra;
```
Comment (chibenwa) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraClusterTest.java]: Missing license

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraClusterTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        cnx1.close();
+        try (CassandraCluster cnx2 = createCluster()) {
+            cnx1.close();
+            assertThatThrownBy(this::createCluster).isInstanceOf(IllegalStateException.class);
+        }
+    }
+
+    @Test
+    void creatingTwoClustersShouldProvideFirstCreationStacktrace() {
+        assertThatThrownBy(this::createCluster).hasStackTraceContaining("setUp");
```
Comment (chibenwa) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraClusterTest.java]: Can we increase the scope of what is being matched? What if junit 5 also have a method matching this? Maybe including the class name could help...

**Comment by rouazana** [comment]
please add a real jira

**Comment by mbaechler** [comment]
build was green before affecting a JIRA number.

**Pull request merged**